HopsanCore/src/ComponentUtilities/TempDirectoryHandle.cpp

```diff
@@ -0,0 +1,263 @@
+ the Hopsan source code root directory.
+
+-----------------------------------------------------------------------------*/
+
+//!
+//! @file   TempDirectoryHandle.cpp
+//! @author Robert Braun <robert.braun@liu.se>
+//! @date   2021-02-02
+//!
+//! @brief Contains a temp directory handle class
+//!
+//$Id$
+
+
+#include "ComponentUtilities/num2string.hpp"
+
+#include <iostream>
+#if __cplusplus >= 201703L
+#include <filesystem>
+#endif
+#if __cplusplus >= 201103L
+#include <random>
+#endif
+#include <sstream>
+#include <iomanip>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <cstring>
+#include <stdio.h>
+#include <stdlib.h>
+#ifdef _WIN32
+#include <windows.h>
+#include <direct.h>
+#include <shellapi.h>
+#include <strsafe.h>
+#else
+#include <dirent.h>
+#include <unistd.h>
+#endif
+
+
+
+#include "ComponentUtilities/TempDirectoryHandle.h"
+
+using namespace hopsan;
+
+TempDirectoryHandle::TempDirectoryHandle(HString prefix)
+{
+        mPath = getTempDirectory();
+        mPath.append("/Hopsan/"+prefix);
+        mPath.append(generateRandomNumericString());
+
+#if __cplusplus >= 201703L
+        mPath = std::filesystem::create_directory(mPath.c_str());
+#else
+        mIsValid = createDirectory(mPath);
+#endif
+}
+
+TempDirectoryHandle::~TempDirectoryHandle()
+{
+    std::cout << "Test\n";
+    removeDirectory(mPath);
+}
+
+HString TempDirectoryHandle::path()
+{
+    return mPath;
+}
+
+bool TempDirectoryHandle::isValid()
+{
+    return mIsValid;
+}
+
+
+HString TempDirectoryHandle::getTempDirectory() {
+#if __cplusplus >= 201703L
+        mPath = std::filesystem::temp_directory_path().c_str();
+#else
+#ifdef _WIN32
+    char tempdirbuff[MAX_PATH+1];
+    GetTempPathA(MAX_PATH+1, tempdirbuff);
+    return HString(tempdirbuff);
+#else
+    char * val = getenv("TMPDIR");
+    if(val != nullptr) {
+        return val;
+    }
+    val = getenv("TMP");
+    if(val != nullptr) {
+        return val;
+    }
+    val = getenv("TEMP");
+    if(val != nullptr) {
+        return val;
+    }
+    val = getenv("TEMPDIR");
+    if(val != nullptr) {
+        return val;
+    }
+    return "/tmp";
+#endif
+#endif
+}
+
+HString TempDirectoryHandle::generateRandomNumericString()
+{
+#if false//__cplusplus >= 201103L
+        std::random_device rd;
+        std::default_random_engine gen(rd());
+        std::uniform_int_distribution<> distrib(0, 999999999);
+        int random = distrib(gen);
+#else
+        int random = rand() % 1000000000;
+#endif
+        std::stringstream ss;
+        ss << std::setw(6) << std::setfill('0') << random;
+        return ss.str().c_str();
+}
+
+bool TempDirectoryHandle::createDirectory(HString path)
+{
+#if __cplusplus >= 201703L
+        return std::filesystem::create_directory(mPath.c_str());
+#else
+    HString currentLevel = "";
+    std::string level;
+    if(!path.empty() && path[0] == '/') {
+        //path.erase(0,1);
+    }
+    std::stringstream ss(path.c_str());
+
+    while(std::getline(ss, level, '/')) {
```

Comment:
Will / work on windows, or do you need to check for \

Reply:
I have tested it on Windows and confirmed that it works.